### PR TITLE
added install/global/monitoring target

### PR DESCRIPTION
**Pull request by jhellar**
https://issues.redhat.com/browse/MGDSTRM-166

### CODE REVIEW

File: install/Makefile [hunk 1]
```diff
@@ -4,9 +4,29 @@ KAFKA_CLUSTER_NAMESPACE ?= openshift-kafka-cluster
...
 GLOBAL_MONITORING_NAMESPACE ?= kafka-monitoring-global
 RESOURCE_FILES_DIR ?= ./resources
+BUILD_DIR ?= ./build
 
 .PHONY: install/global/monitoring
 install/global/monitoring:
+  # create yaml files
+	@ rm -rf $(BUILD_DIR)/grafana
+	@ mkdir -p $(BUILD_DIR)/grafana
+	@ cp $(RESOURCE_FILES_DIR)/grafana/operatorgroup.yaml $(BUILD_DIR)/grafana/
```
Comment (damienomurchu) [inline install/Makefile:14]: Should we fetch the resources from operatorhub, ie `wget https://operatorhub.io/install/grafana-operator.yaml`

File: install/Makefile [hunk 2]
```diff
@@ -4,9 +4,29 @@ KAFKA_CLUSTER_NAMESPACE ?= openshift-kafka-cluster
...
+	@ oc new-project $(PER_CLUSTER_MONITORING_NAMESPACE)
+
+	# create operatorgroup
+	@ oc apply -f $(BUILD_DIR)/grafana/operatorgroup.yaml
+
+	# # create subscription
+	@ oc apply -f $(BUILD_DIR)/grafana/sub.yaml
+
+	# # create grafana CR
+	@ oc apply -f $(BUILD_DIR)/grafana/grafana.yaml
```
Comment (damienomurchu) [inline install/Makefile:29]: Perhaps remove the build files as the last step?

**Comment by damienomurchu** [comment]
Looks good, left some small comments

File: install/Makefile (hunk 1 above)
Comment (david-martin) [reply install/Makefile:14]: I'd prefer to check them in here for a couple reasons:
- avoid downtime issues with operatorhub
- allow easier tweaking of them in the short term

File: install/Makefile [hunk 3]
```diff
@@ -4,9 +4,29 @@ KAFKA_CLUSTER_NAMESPACE ?= openshift-kafka-cluster
...
 
 .PHONY: install/global/monitoring
 install/global/monitoring:
+  # create yaml files
+	@ rm -rf $(BUILD_DIR)/grafana
+	@ mkdir -p $(BUILD_DIR)/grafana
+	@ cp $(RESOURCE_FILES_DIR)/grafana/operatorgroup.yaml $(BUILD_DIR)/grafana/
+	@ cp $(RESOURCE_FILES_DIR)/grafana/sub.yaml $(BUILD_DIR)/grafana/
+	@ cp $(RESOURCE_FILES_DIR)/grafana/grafana.yaml $(BUILD_DIR)/grafana/
+	@ sed -i 's/<PER_CLUSTER_MONITORING_NAMESPACE>/$(PER_CLUSTER_MONITORING_NAMESPACE)/' $(BUILD_DIR)/grafana/*.yaml
```
Comment (david-martin) [inline install/Makefile:17]: Grafana should be installed into a non services specific namespace
e.g. `managed-services-monitoring`(nothing kafka specific)

File: install/Makefile [hunk 4]
```diff
@@ -4,9 +4,29 @@ KAFKA_CLUSTER_NAMESPACE ?= openshift-kafka-cluster
 PER_CLUSTER_MONITORING_NAMESPACE ?= kafka-monitoring-cluster
 GLOBAL_MONITORING_NAMESPACE ?= kafka-monitoring-global
```
Comment (david-martin) [inline install/Makefile:5]: These vars are a little confusing.

There would be 2 namespaces where monitoring stuff goes.

- managed-services-monitoring (1 per OSD cluster, contains grafana (& prometheus in the short term))
- the strimzi operator namespace (Put strimzi/kafka specific resources)

**Review by david-martin — APPROVED** [review]

**Review by david-martin — CHANGES_REQUESTED** [review]
On hold pending #5

File: install/Makefile (hunk 2 above)
Comment (jhellar) [reply install/Makefile]: Makes sense, moved

File: install/Makefile (hunk 3 above)
Comment (jhellar) [reply install/Makefile]: This is renamed now

File: install/Makefile (hunk 4 above)
Comment (jhellar) [reply install/Makefile]: This should be renamed now as well

**Comment by pb82** [comment]
@jhellar #5 is merged now, can we also get this merged?

**Review by damienomurchu — APPROVED** [review]

**Comment by damienomurchu** [comment]
@david-martin anything on your end holding back this from being merged now? All PR feedback looks to have been addressed

**Comment by david-martin** [comment]
👀

File: install/Makefile [hunk 5]
```diff
@@ -1,13 +1,34 @@
...
-PER_CLUSTER_MONITORING_NAMESPACE ?= kafka-monitoring-cluster
-GLOBAL_MONITORING_NAMESPACE ?= kafka-monitoring-global
+PER_CLUSTER_MONITORING_NAMESPACE ?= managed-services-monitoring
+GLOBAL_MONITORING_NAMESPACE ?= managed-services-monitoring-global
 KAFKA_PVC_SIZE ?= 10Gi
 RESOURCE_FILES_DIR ?= ./resources
 BUILD_DIR := build
 
 .PHONY: install/global/monitoring
 install/global/monitoring:
```
Comment (pb82) [inline install/Makefile:10]: should this be `install/cluster/monitoring`

File: install/Makefile [hunk 6]
```diff
@@ -1,13 +1,34 @@
...
+  # create yaml files
+	@ mkdir -p ./$(BUILD_DIR)/monitoring-cluster
+	@ cp $(RESOURCE_FILES_DIR)/monitoring-cluster/operatorgroup.yaml ./$(BUILD_DIR)/monitoring-cluster/
+	@ cp $(RESOURCE_FILES_DIR)/monitoring-cluster/sub.yaml ./$(BUILD_DIR)/monitoring-cluster/
+	@ cp $(RESOURCE_FILES_DIR)/monitoring-cluster/grafana.yaml ./$(BUILD_DIR)/monitoring-cluster/
+	@ sed -i 's/<PER_CLUSTER_MONITORING_NAMESPACE>/$(PER_CLUSTER_MONITORING_NAMESPACE)/' ./$(BUILD_DIR)/monitoring-cluster/*.yaml
+	
+	# create namepsace
+	@ oc new-project $(PER_CLUSTER_MONITORING_NAMESPACE)
```
Comment (pb82) [inline install/Makefile:19]: This fails on second run when the namespace already exists. You can use this to ignore the error:

`@ oc new-project $(PER_CLUSTER_MONITORING_NAMESPACE) | true`

**Review by pb82 — APPROVED** [review]
Verified locally

**Pull request merged**